Earthfile

```diff
@@ -1092,3 +1177,67 @@ bump-repositories:
     RUN yq eval ".repositories[0] |= . * { \"reference\": \"${REPO_AMD64}\" }" -i framework-profile.yaml
     RUN yq eval ".repositories[1] |= . * { \"reference\": \"${REPO_ARM64}\" }" -i framework-profile.yaml
     SAVE ARTIFACT framework-profile.yaml AS LOCAL framework-profile.yaml
+
+# Installs the needed bits for "standard" images (the provider ones)
+PROVIDER_INSTALL:
+    COMMAND
+
+    ARG PROVIDER_KAIROS_BRANCH
+
+    COPY +luet/luet /usr/bin/luet
+
+    IF [[ "$PROVIDER_KAIROS_BRANCH" = "" ]] # Install with luet (released versions of the binary)
+      # If base image does not bundle a luet config use one
+      # TODO: Remove this, use luet config from base images so they are in sync
+      IF [ ! -e "/etc/luet/luet.yaml" ]
+          COPY framework-profile.yaml /etc/luet/luet.yaml
+      END
+
+      # We don't specify a version. To bump, just change what the latest version
+      # in the repository is.
+      RUN luet install -y system/provider-kairos
+    ELSE # Install from a branch
+      COPY github.com/kairos-io/provider-kairos:$PROVIDER_KAIROS_BRANCH+build-kairos-agent-provider/agent-provider-kairos /system/providers/agent-provider-kairos
+      RUN ln -s /system/providers/agent-provider-kairos /usr/bin/kairos
+    END
+
+INSTALL_NOHANG:
+    COMMAND
+
+    ARG FLAVOR
+
+    # Install nohang
+    IF [ "$FLAVOR" = "opensuse-leap" ] || [ "$FLAVOR" = "opensuse-leap-arm-rpi" ]
+      RUN zypper ref && zypper in -y nohang
+    ELSE IF [ "$FLAVOR" = "alpine-ubuntu" ] || [ "$FLAVOR" = "alpine-opensuse-leap" ] || [ "$FLAVOR" = "alpine-arm-rpi" ]
+      RUN apk add grep
```

Comment:
:confused: So since alpine doesn't have `nohang` we decided to install `grep` instead ? :D

Reply:
We only enable this on systemd: https://github.com/kairos-io/kairos/blob/master/overlay/files/system/oem/09_services.yaml#L21-L22

So the question is, how does this work on systemd flavors that don't have it?
(fedora, rockylinux, almalinux)

Reply:
lol